src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs

```diff
@@ -142,15 +143,36 @@ protected override void Dispose(bool finalize)
                 document.CloseTextView();
             }
 
-            var exceptions = ExportProvider.GetExportedValue<TestExtensionErrorHandler>().GetExceptions();
+            var exceptions = Flatten(ExportProvider.GetExportedValue<TestExtensionErrorHandler>().GetExceptions());
 
-            if (exceptions.Count == 1)
+            if (exceptions.Count > 0)
             {
-                throw exceptions.Single();
-            }
-            else if (exceptions.Count > 1)
-            {
-                throw new AggregateException(exceptions);
+                var messageBuilder = new StringBuilder();
+                messageBuilder.AppendLine(
+$@"{exceptions.Count} exception(s) were thrown during test.
+Note: exceptions may have been thrown by another test running concurrently with
+this test.  This can happen with any tests that share the same ExportProvider.
+Examining individual exception stacks may help reveal the original test and source 
+of the problem.");
+
+                messageBuilder.AppendLine();
+                for (int i = 0; i < exceptions.Count; i++)
```

Comment:
AggregateException already does roughly this: https://source.dot.net/#System.Private.CoreLib/src/System/AggregateException.cs,434